src/proxy/inbound_passthrough.rs

```diff
@@ -64,21 +66,36 @@ impl InboundPassthrough {
     }
 
     pub(super) async fn run(self) {
+        let (sub_drain_signal, sub_drain) = drain::channel();
+        let deadline = self.pi.cfg.self_termination_deadline;
```

Comment:
At some point we may want separate vars rather than reusing this for everything - it's overloaded with different meanings already.

Also, we probably should have a more robust form of shutdown signaling with a _single_ channel, rather than creating 2-3 different shutdown channels and using them in concert.

That's probably better handled as part of https://github.com/istio/ztunnel/issues/899 tho.